### PR TITLE
feat: fast musdb loader

### PR DESCRIPTION
**Pull request by yoyololicon**
Copy-past from my [repo](https://github.com/yoyololicon/music-demixing-challenge-ismir-2021-entry/blob/2f08f9b8050b7b2fc360c625ec5a939825156585/dataset.py#L26), with all soundfile audio readers replaced by torchaudio.

### CODE REVIEW

File: AudioLoader/Music.py [hunk 1]
```diff
@@ -694,6 +699,145 @@ def available_groups(self, group):
...
+            if self.random_track_mix and self.random:
+                x = sum(stems)
+            else:
+                x, _ = torchaudio.load(
+                    os.path.join(folder_name, 'mixture.wav'), frame_offset=chunk_start, num_frames=self.segment)
+                if x.shape[1] < self.segment:
+                    x = F.pad(x.unsqueeze(1),
+                              (0, self.segment - x.shape[1])).squeeze(1)
+
+        y = np.stack(stems)
```
Comment (yoyololicon) [inline AudioLoader/Music.py:839]: Sorry, I forgot to change this line to ```y = torch.stack(stems)```. You can fix it yourself or I can do it later.

Comment (KinWaiCheuk) [reply AudioLoader/Music.py:839]: Don't worry, I can fix it.

Comment (KinWaiCheuk) [reply AudioLoader/Music.py:839]: Actually, even without fixing it, `Dataloader` automatically converts them into torch tensors. Do you know if returning torch tensors inside `__getitem__` is faster than numpy arrays?

Comment (yoyololicon) [reply AudioLoader/Music.py:839]: Not sure though. The problem is that `torchaudio.load` is used so `stems` are all torch tensors.

File: AudioLoader/Music.py [hunk 2]
```diff
@@ -697,3 +702,145 @@ def available_groups(self, group):
...
+    def _get_random_track_idx(self):
+        return random.randrange(len(self.tracks))
+
+    def _get_random_start(self, length):
+        return random.randrange(length - self.segment + 1)
+
+    def _get_track_from_chunk(self, index):
+        track_idx = np.digitize(index, self.cum_chunks)
+        if track_idx > 0:
+            chunk_start = (index - self.cum_chunks[track_idx]) * self.segment
```
Comment (KinWaiCheuk) [inline AudioLoader/Music.py:794]: When `track_idx`>0, `chunk_start` is always negative.
Maybe `(self.cum_chunks[track_idx] - index) * self.segment` is the correct one?

Comment (yoyololicon) [reply AudioLoader/Music.py:794]:
```suggestion
            chunk_start = (index - self.cum_chunks[track_idx-1]) * self.segment
```

**Pull request merged**